ladybug_geometry/geometry3d/polyface.py

```diff
@@ -106,6 +107,46 @@ def __init__(self, vertices, face_indices, edge_information=None):
         self._center = None
         self._area = None
         self._volume = None
+        self._tolerance = None
+        self._angle_tolerance = None
+
+    @classmethod
+    def from_dict(cls, data):
+        """Create a Face3D from a dictionary.
+
+        Args:
+            data: {"faces": [
+                    {"boundary": [{"x": 0, "y": 0, "z": 0}, {"x": 10, "y": 0, "z": 0},
+                                 {"x": 0, "y": 10, "z": 0}],
+                    "plane": {"n": {"x": 0, "y": 0, "z": 1},
+                              "o": {"x": 0, "y": 0, "z": 0},
+                              "x": {"x": 1, "y": 0, "z": 0}}},
+                    {"boundary": [{"x": 10, "y": 0, "z": 0}, {"x": 10, "y": 10, "z": 0},
+                                 {"x": 0, "y": 10, "z": 0}],
+                    "plane": {"n": {"x": 0, "y": 0, "z": 1},
+                              "o": {"x": 0, "y": 0, "z": 0},
+                              "x": {"x": 1, "y": 0, "z": 0}}}
+                ],
+                "tolerance": None
```

Comment:
missing `angle_tolerance`

Reply:
As noted above, I think we need a larger edit that takes the tolerance out of the object property so let me come back to this.